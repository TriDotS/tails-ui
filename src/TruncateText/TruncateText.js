import React, { PureComponent } from 'react'

import PropTypes from 'prop-types'

export default class TruncateText extends PureComponent {
  static propTypes = {
    children: PropTypes.string,
    wordCount: PropTypes.number
  }

  static defaultProps = {
    wordCount: 10
  }

  state = {
    showMore: false
  }

  toggleShowMore = () => this.setState({ showMore: !this.state.showMore })

  render() {
    const { showMore } = this.state
    const { children, wordCount } = this.props

    const spliced = children
      .split(' ')
      .splice(0, wordCount)
      .join(' ')

    const truncated =
      spliced.split('').lastIndexOf(',') === spliced.length - 1
        ? spliced.slice(0, spliced.length - 1)
        : spliced

    return (
      <span>
        {showMore ? children : truncated}
        {children.split(' ').length > wordCount && (
          <span onClick={() => this.toggleShowMore()}>
            {!showMore ? <span>...more</span> : <span>...less</span>}
          </span>
        )}
      </span>
    )
  }
}
