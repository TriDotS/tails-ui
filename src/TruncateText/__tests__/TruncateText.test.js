import React from 'react'
import TruncateText from '../TruncateText'
import { render } from 'react-dom'
import renderer from 'react-test-renderer'

it('renders without crashing', () => {
  const div = document.createElement('div')
  render(
    <TruncateText wordCount={10}>
      This is some text that should be truncated because its too long to display
      with a word count of 10
    </TruncateText>,
    div
  )
})

test('t snapshot', () => {
  const tree = renderer
    .create(
      <TruncateText wordCount={10}>
        This is some text that should be truncated because its too long to
        display with a word count of 10
      </TruncateText>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
