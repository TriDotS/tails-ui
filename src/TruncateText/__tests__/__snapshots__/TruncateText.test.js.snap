// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`t snapshot 1`] = `
<span>
  This is some text that should be truncated because its
  <span
    onClick={[Function]}
  >
    <span>
      ...more
    </span>
  </span>
</span>
`;
